client/api/src/execution_extensions.rs

```diff
@@ -33,34 +33,34 @@ use sp_runtime::{
 	generic::BlockId,
 	traits,
 };
-use sp_state_machine::{ExecutionManager, DefaultHandler};
-pub use sp_state_machine::ExecutionStrategy;
+use sp_state_machine::{ExecutionManager, ExecutionStrategy, ExecutionConfig, DefaultHandler};
 use sp_externalities::Extensions;
 use parking_lot::RwLock;
 
-/// Execution strategies settings.
+/// Execution configurations, per operation type.
 #[derive(Debug, Clone)]
-pub struct ExecutionStrategies {
-	/// Execution strategy used when syncing.
-	pub syncing: ExecutionStrategy,
-	/// Execution strategy used when importing blocks.
-	pub importing: ExecutionStrategy,
-	/// Execution strategy used when constructing blocks.
-	pub block_construction: ExecutionStrategy,
-	/// Execution strategy used for offchain workers.
-	pub offchain_worker: ExecutionStrategy,
-	/// Execution strategy used in other cases.
-	pub other: ExecutionStrategy,
+pub struct ExecutionConfigs {
+	/// Execution config used when syncing.
+	pub syncing: ExecutionConfig,
+	/// Execution config used when importing blocks.
+	pub importing: ExecutionConfig,
+	/// Execution config used when constructing blocks.
+	pub block_construction: ExecutionConfig,
+	/// Execution config used for offchain workers.
+	pub offchain_worker: ExecutionConfig,
+	/// Execution config used in other cases.
+	pub other: ExecutionConfig,
 }
 
-impl Default for ExecutionStrategies {
-	fn default() -> ExecutionStrategies {
-		ExecutionStrategies {
-			syncing: ExecutionStrategy::NativeElseWasm,
-			importing: ExecutionStrategy::NativeElseWasm,
-			block_construction: ExecutionStrategy::AlwaysWasm,
-			offchain_worker: ExecutionStrategy::NativeWhenPossible,
-			other: ExecutionStrategy::NativeElseWasm,
+impl Default for ExecutionConfigs {
+	fn default() -> ExecutionConfigs {
+		// TODO: this should use our defaults consts.
+		ExecutionConfigs {
```

Comment:
Would have been good if the constants defined in `sc-cli` (e.g. `DEFAULT_EXECUTION_SYNCING`, `DEFAULT_CONTEXT_SYNCING`) were moved to `sc-api` or sth to be usable in more places.